server/src/main/java/org/jetbrains/bsp/bazel/server/sync/BazelProjectMapper.kt

```diff
@@ -89,15 +89,22 @@ class BazelProjectMapper(
         target.dependenciesList.map { Label(it.id) }
 
     private fun inferLanguages(target: TargetInfo): Set<Language> =
-        target.sourcesList.flatMap { source: FileLocation ->
-            Language.all().filter { isLanguageFile(source, it) }
-        }.toHashSet()
+        if (target.sourcesList.isEmpty()) {
+            Language.all().filter { isBinaryTargetOfLanguage(target.kind, it) }.toHashSet()
```

Comment:
i think that we can actually call it alway, and for libraries and tests as well

Reply:
Libraries and tests both have source files, so sources-based language detecting works better for them (especially for multi-language ones).